feat(perf-vitals): Add INP to allowed measurements

k-fish opened the pull request:
### Summary
This allows the `INP` measurement to be sent w/o counting towards the measurement limit.

k-fish committed:
feat(perf-vitals): Add INP to allowed measurements

This allows the `INP` measurement to be sent w/o counting towards the measurement limit.

jjbayer reviewed (approved):
I see this string already exists in the indexer table, otherwise we could have added it to the set of static strings as well. https://github.com/getsentry/sentry/blob/3d109cad5ca7e5083592a0419aa91fdbbcd06c08/src/sentry/sentry_metrics/indexer/strings.py#L48

k-fish committed:
Add INP to tests

k-fish commented:
Tests need https://github.com/getsentry/relay/pull/1487 to be merged before they pass I think, since it's currently getting `none` instead of being cast to milliseconds.

k-fish committed:
Merge branch 'master' into feat/perf-vitals-add-inp-to-allowed-measurements

k-fish committed:
Add measurement to the prefixed transaction metrics names

We may already have the metric indexed as a non-known metric, since people have already sent it, but regardless we should have it in the list of our standard metrics names for ST etc.

jjbayer commented on src/sentry/sentry_metrics/indexer/strings.py:
```diff
@@ -71,6 +71,7 @@
     "d:transactions/breakdowns.span_ops.ops.browser@millisecond": PREFIX + 122,
     "d:transactions/breakdowns.span_ops.ops.resource@millisecond": PREFIX + 123,
     "d:transactions/breakdowns.span_ops.ops.ui@millisecond": PREFIX + 124,
+    "d:transactions/measurements.inp@millisecond": PREFIX + 125,
```
Adding this string means that data collected before this change will be effectively lost for this metric, as any org-specific string index for this metric will now be obscured by the index defined here. Afaict this currently affects two orgs.

k-fish replied:
Thanks for taking a look again Joris 👍 

I guess I'm wondering a bit about the purpose these predefined metrics serve if moving forward we'll likely have to have mixed pre-defined (these static ones) and ones dynamically created already. Of course changing the test works too, but is this list now frozen in time? I imagine as more and more people use measurements, most measurement names we'll come with will already have been sent in, thereby making this list no-longer updatable.

k-fish replied:
I'll do your recommendation of changing the test code, for now though, that is just a question moving forward for this list in the metrics indexer.

jjbayer replied:
> is this list now frozen in time? I imagine as more and more people use measurements, most measurement names we'll come with will already have been sent in

Yes I think that is an accurate assessment, looking at

https://github.com/getsentry/sentry/blob/9f4ca288acbb722a39650f5c04efdaa81ae46a58/src/sentry/sentry_metrics/indexer/strings.py#L18-L19

If at some point we want to reduce storage cost by consolidating duplicate strings, I guess we'll need a migration strategy to migrate the data behind a specific string from per-org to shared. cc @onewland

k-fish committed:
Switch to modify the test instead
